wgengine/magicsock/magicsock.go

```diff
@@ -985,6 +1006,16 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 		return nil, err
 	}
 
+	portmap := make(chan netaddr.IPPort, 1)
+	go func() {
+		if ext, err := c.portMapper.CreateOrGetMapping(ctx); err == nil {
+			portmap <- ext
+		} else if !portmapper.IsNoMappingError(err) {
+			c.logf("portmapper: %v", err)
+		}
+		close(portmap)
```

Comment:
maybe just close it in the failure path? It's a little redundant in normal operation.

Reply:
This is a bit confusing, but having it there is necessary if there was a `NoMappingError`, otherwise the other branch won't be hit.